wdl-lint/src/rules/command_shellcheck.rs

```diff
@@ -0,0 +1,357 @@
+                    decls.insert(bash_var.clone());
+                    let mut expansion = String::from("\"$");
+                    expansion.push_str(&bash_var);
+                    expansion.push('"');
+                    sanitized_command.push_str(&expansion);
+                }
+            });
+        } else {
+            // TODO add a diagnostic explaining why this is being skipped?
+            return;
+        }
+
+        // get leading whitespace so we can add it to each span
+        // TODO there may be a more elegant way to do this
+        let leading_whitespace = {
+            if let Some(first_part) = section.parts().next() {
+                match first_part {
+                    CommandPart::Text(text) => {
+                        let text_str = text
+                            .as_str()
+                            .strip_prefix("\n")
+                            .unwrap_or_else(|| text.as_str());
+                        count_leading_whitespace(text_str)
+                    }
+                    CommandPart::Placeholder(pholder) => {
+                        let pholder_str = &pholder.syntax().text().to_string();
+                        count_leading_whitespace(
+                            pholder_str.strip_prefix("\n").unwrap_or(pholder_str),
+                        )
+                    }
+                }
+            } else {
+                0
+            }
+        };
+
+        // Map each actual line of the command to its corresponding
+        // CommandPart and start position
+        let mut line_map = HashMap::new();
+        let mut on_same_line = false;
+        let mut line_num = 0usize;
+        for part in section.parts() {
+            match part {
+                CommandPart::Text(ref text) => {
+                    let mut last_line = 0usize;
+                    for (_, start, _) in lines_with_offset(text.as_str()) {
+                        // Check to see if we are still on the previous line
+                        // This happens after we encounter a placeholder
+                        if on_same_line {
+                            on_same_line = false;
+                            last_line = last_line.saturating_sub(1);
+                        }
+                        line_map.insert(line_num + last_line, (part.clone(), start));
+                        last_line += 1;
+                    }
+                    line_num += last_line;
+                }
+                CommandPart::Placeholder(_) => {
+                    on_same_line = true;
+                }
+            }
+        }
+
+        match run_shellcheck(&sanitized_command) {
+            Ok(comments) => {
+                for comment in comments {
+                    // skip declarations that shellcheck is unaware of
+                    if comment.code == 2154
+                        && decls.contains(comment.message.split_whitespace().next().unwrap_or(""))
```

Comment:
It might be good to assign what's in the parentheses to a variable first with a descriptive name. I'm not sure that this line is checking from first glance.